File: Questions/263. ugly-number.c
// https://leetcode.com/problems/ugly-number

bool isUgly(int num) {
    if (num==1) return true;
    if (num==0) return false;
    while(num!=1)
    {
        if(num%2==0 || num%3==0 || num%5==0)
        {
            if(num%2==0) num/=2;
            if(num%3==0) num/=3;
            if(num%5==0) num/=5;
        }
        else if(num!=1)
            return false;
    }
    return true;
}

/* 有點神
for (int i=2; i<6 && num; i++)
    while (num % i == 0)
        num /= i;
return num == 1;
*/

File: Problems/263.ugly-number.java
// https://leetcode.com/problems/ugly-number

class Solution {
    public boolean isUgly(int n) {
        if(n==1)
            return true;
        
        int prev=0;
        
        while(n!=prev){
            prev=n;
            while(n%2==0)
                n/=2;
            while(n%5==0)
                n/=5;
            while(n%3==0)
                n/=3;
            
            if(n==1)
                return true;
        }
        
        return false;
        
    }
}